src/system/sys_profiler.cc

```diff
@@ -549,7 +550,10 @@ void Profiler::add_timer_info(ReduceFunctor reduce, property_tree::ptree* holder
     // using operation MPI_MIN, so if some processor does not contain some 
     // time-frame other processors will forget this time-frame (information lost)
     int child_timers[ Timer::max_n_childs];
-    MPI_Allreduce(&timer.child_timers, &child_timers, Timer::max_n_childs, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
+    if (Application::petsc_initialized) {
+        MPI_Allreduce(&timer.child_timers, &child_timers, Timer::max_n_childs, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
```

Comment:
Sorry, this is not solution. method output() just should not use MPI at all. I disable this check, it would be better to implement our own MPI reduce operation for the tree sooner then planed.